res/gamedata/shaders/gl/shadow.h

```diff
@@ -155,35 +156,36 @@ float shadow_extreme_quality( float3 tc )
     {
         for( int col = -FS2; col <= FS2; col +=2 )
         {
-            float4 d4;
-            float  b;
+          float4 d4;
```

Comment:
You are using incorrect tab-size of 2, it should be 4.
And the body of this cycle still has the extra diff because original shaders use mixed tabs and spaces, and probably your code formatter replaced tabs with spaces.